src/ProfSpline/ObservableI.h

```diff
@@ -0,0 +1,41 @@
+// Class to handle all observables as a whole
+// This is a interface class
+
+#ifndef _OBSERVABLE_I_
+#define _OBSERVABLE_I_
+#include "Framework/Algorithm/Algorithm.h"
+#include "Framework/EventGen/EventRecord.h"
+#include <string>
+#include <vector>
+namespace genie {
+namespace rew {
+/// \uml{note Made this an interface to handle the possibliy of
+/// different choice of different observables.
+/// }
+class RwgKineSpace : public Algorithm {
+public:
+  // ObservableI(std::string name) : Algorithm(name) {}
+  using Algorithm::Algorithm;
```

Comment:
This is can be an issue in the future. Exposed the constructors only in protected area.